engine/gamesys/src/gamesys/components/comp_particlefx.cpp

```diff
@@ -282,6 +282,8 @@ namespace dmGameSystem
             pfx_world->m_ConstantBuffers[ro_index] = dmRender::NewNamedConstantBuffer();
         }
         ro.m_ConstantBuffer = pfx_world->m_ConstantBuffers[ro_index];
+
+        dmRender::ClearNamedConstantBuffer(ro.m_ConstantBuffer);
         SetRenderConstants(ro.m_ConstantBuffer, first->m_RenderConstants, first->m_RenderConstantsSize);
```

Comment:
The named constant buffer (`pfx_world->m_ConstantBuffers[ro_index]`) is reused.
So if the "ro_index" changes, it's possible you get a buffer that already contains constants.
The fix (also used in other components), is to clear the constants from the buffer (/internally a hash table) before adding new constants.